packages/material-ui-lab/src/internal/pickers/PickersModalDialog.tsx

```diff
@@ -17,6 +17,13 @@ export interface ExportedPickerModalProps {
    * @default 'Cancel'
    */
   cancelText?: React.ReactNode;
+  /**
+   * The props used for each slot inside.
+   * @default {}
+   */
+  componentsProps?: {
+    clearButton?: any; // TODO: Consider changing 'any' to 'Partial<ButtonProps>'.
```

Comment:
What should we do here? Do we allow the developer to just pass any prop or attribute via `clearButton`?

Reply:
@michaldudak can help with this

Reply:
@alon-glater The pattern we use is to restrict the componentsProps to the props of the rendered element but allow for overrides.
To be precise:
```tsx
componentsProps?: {
    clearButton?: React.ComponentPropsWithRef<Button> & PickersModalDialogComponentsPropsOverrides;
}
```

where PickersModalDialogComponentsPropsOverrides is an exported empty interface:

```tsx
export interface PickersModalDialogComponentsPropsOverrides {}
```

This way, developers can use all the props accepted by `Button` out of the box and still be able to add custom ones with the help of module augmentation.